remix-analyzer/src/solidity-analyzer/modules/gasCosts.ts

```diff
@@ -1,72 +1,136 @@
-      if (fallback !== undefined) {
-        if (fallback === null || parseInt(fallback) >= 2100 || fallback === 'infinite') {
-          report.push({
-            warning: `Fallback function of contract ${contract.name} requires too much gas (${fallback}). 
-            If the fallback function requires more than 2300 gas, the contract cannot receive Ether.`
-          })
+      else 
+        signature = node.name + '()'
+      
+      return {
+        name: node.name,
+        src: node.src,
+        signature: signature
+      }
+    })
+    for (const method of methodsWithSignature) {
+      for (const filename in compilationResults.contracts) {
+        for (const contractName in compilationResults.contracts[filename]) {
+          const contract: CompiledContract = compilationResults.contracts[filename][contractName]
+          const methodGas: Record<string, any> | undefined = this.checkMethodGas(contract, method.signature)
+          if(methodGas && methodGas.isInfinite) {
+            if(methodGas.isFallback) {
+              report.push({
+                warning: `Fallback function of contract ${contractName} requires too much gas (${methodGas.msg}). 
+                If the fallback function requires more than 2300 gas, the contract cannot receive Ether.`,
+                location: method.src
+              })
+            } else {
+              report.push({
+                warning: `Gas requirement of function ${contractName}.${method.name} ${methodGas.msg}. 
+                If the gas requirement of a function is higher than the block gas limit, it cannot be executed.
+                Please avoid loops in your functions or actions that modify large areas of storage
+                (this includes clearing or copying arrays in storage)`,
+                location: method.src
+              })
+            } 
+          } else continue
         }
       }
+    }
+    return report
+  }
 
-      for (var functionName in contract.object.evm.gasEstimates.external) {
-        if (functionName === '') {
-          continue
-        }
-        const gas: string = contract.object.evm.gasEstimates.external[functionName]
-        const gasString: string = gas === null ? 'unknown or not constant' : 'high: ' + gas
-        if (gas === null || parseInt(gas) >= 3000000 || gas === 'infinite') {
-          report.push({
-            warning: `Gas requirement of function ${contract.name}.${functionName} ${gasString}. 
-            If the gas requirement of a function is higher than the block gas limit, it cannot be executed.
-            Please avoid loops in your functions or actions that modify large areas of storage
-            (this includes clearing or copying arrays in storage)`
-          })
+  // To create the method signature similar to contract.evm.gasEstimates.external object
+  // For address payable, return address 
+  private getSplittedTypeDesc(node: FunctionDefinitionAstNode, contracts: CompiledContractObj): string[] {
```

Comment:
does this handle nested struct? seems so but wanted confirmation

Reply:
I am working on its unit testing.

Reply:
I have added them.